core/src/db/engine/ExecutionEngineImpl.cpp

```diff
@@ -448,7 +448,13 @@ ExecutionEngineImpl::Load(bool to_cache) {
             try {
                 segment::SegmentPtr segment_ptr;
                 segment_reader_ptr->GetSegment(segment_ptr);
-                auto status = segment_reader_ptr->LoadVectorIndex(location_, segment_ptr->vector_index_ptr_);
+                if (index_type_ == EngineType::FAISS_IVFFLAT || index_type_ == EngineType::HNSW ||
```

Comment:
need add a API in IndexType.cpp: 
bool IndexSupportOffset(std::string index_type)

Reply:
index_type_ is enum EngineType. Added method in ExecutionEngineImpl.cpp, similar to 
`bool IsBinaryIndexType(knowhere::IndexType type)`
in Line#90.